components/AmountInput/AmountInput.js

```diff
@@ -24,6 +24,16 @@ function AmountInput({
   onChange,
 }) {
   const viewport = useViewport()
+
+  // Super ugly Next.js workaround to let us have differences between SSR & client
+  const [isCompact, setIsCompact] = useState(false)
+  const smallLayout = viewport.below(414)
+  useEffect(() => {
+    setTimeout(() => {
+      setIsCompact(smallLayout)
+    }, 0)
+  }, [smallLayout])
```

Comment:
The props of the components of the SSR and client versions need to be exactly the same for Next.js to be able to re-conciliate the two. An easy solution would be to disable SSR, but we would lose the main reason why we are using Next.js in the first place.

I’m wondering if a solution could be to modify `useViewport()` so that it always starts with a `0, 0` viewport, then updates… but it means rendering twice by default, and probably creating a visible jump as well.

See https://github.com/zeit/next.js/issues/7322